master/buildbot/data/schedulers.py

```diff
@@ -81,13 +90,30 @@ class Scheduler(base.ResourceType):
     plural = "schedulers"
     endpoints = [SchedulerEndpoint, SchedulersEndpoint]
     keyFields = ['schedulerid']
+    eventPathPatterns = """
+        /schedulers/:schedulerid
+    """
 
     class EntityType(types.Entity):
         schedulerid = types.Integer()
         name = types.String()
+        enabled = types.Boolean()
         master = types.NoneOk(masters.Master.entityType)
     entityType = EntityType(name)
 
+    @defer.inlineCallbacks
+    def generateEvent(self, schedulerid, event):
+        scheduler = yield self.master.data.get(('schedulers', str(schedulerid)))
+        self.produceEvent(scheduler, event)
+
+    @base.updateMethod
+    @defer.inlineCallbacks
+    def schedulerSetEnabled(self, schedulerid, enabled_value):
```

Comment:
I would rather call this API 

    def schedulerEnable(self, schedulerid, enable=True)